unittest/client/src/ilm_common_unittests.cpp

```diff
@@ -0,0 +1,374 @@
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <gtest/gtest.h>
+#include "ilm_common.h"
+#include "ilm_common_platform.h"
+#include "client_api_fake.h"
+
+extern "C"
+{
+    FAKE_VALUE_FUNC(ilmErrorTypes, ilmControl_init, t_ilm_nativedisplay);
+    FAKE_VOID_FUNC(ilmControl_destroy);
+    FAKE_VALUE_FUNC(ilmErrorTypes, ilmControl_registerShutdownNotification, shutdownNotificationFunc, void*);
+}
+
+class IlmCommonTest : public ::testing::Test 
+{
+public:
+    void SetUp()
+    {
+        CLIENT_API_FAKE_LIST(RESET_FAKE);
+        RESET_FAKE(ilmControl_init);
+        RESET_FAKE(ilmControl_destroy);
+        RESET_FAKE(ilmControl_registerShutdownNotification);
+    }
+
+    void TearDown()
+    {
+        if(ilm_isInitialized())
+        {
+            ilm_destroy();
+        }
+    }
+
+    void mock_ilmInitSuccess()
+    {
+        mpp_wlDisplays[0] = (struct wl_display*)&m_wlDisplayFakePointer;
+        SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+
+        mp_ilmErrorType[0] = ILM_SUCCESS;
+        SET_RETURN_SEQ(ilmControl_init, mp_ilmErrorType, MAX_NUMBER);
+    }
+
+    static constexpr uint8_t MAX_NUMBER = 1;
+    uint8_t m_wlDisplayFakePointer = 0;
+    struct wl_display* mpp_wlDisplays [MAX_NUMBER] = {nullptr};
+    ilmErrorTypes mp_ilmErrorType[MAX_NUMBER] = {ILM_FAILED};
+};
+
+/** ================================================================================================
+ * @test_id             ilm_init_cannotGetDisplay
+ * @brief               A failure of ilm_init() when wl_display_connect() returns a null object
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_display_connect() to return a null object
+ *                      -# Calling the ilm_init()
+ *                      -# Verification point:
+ *                         +# ilm_init() must return ILM_FAILED
+ *                         +# wl_display_connect() must be called once time and return nullptr
+ */
+TEST_F(IlmCommonTest, ilm_init_cannotGetDisplay)
+{
+    mpp_wlDisplays[0] = nullptr;
+    SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+    ASSERT_EQ(ILM_FAILED, ilm_init());
+    ASSERT_EQ(wl_display_connect_fake.call_count, 1);
+    ASSERT_EQ(wl_display_connect_fake.return_val_history[0], nullptr);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_init_getFailedOnilmControl_init
+ * @brief               A failure of ilm_init() when ilmControl_init() returns ILM_FAILED
```

Comment:
this I would also rephrase with the similar pattern: Test case of ilm_init() where the internal function ilmControl_init() fails